dinky-admin/src/main/java/org/dinky/service/impl/TaskServiceImpl.java

```diff
@@ -962,15 +965,14 @@ private JobConfig buildJobConfig(Task task) {
                     gatewayConfig.put("userJarParas", jar.getParas());
                     gatewayConfig.put("userJarMainAppClass", jar.getMainClass());
                 } else {
-                    SystemConfiguration systemConfiguration = SystemConfiguration.getInstances();
-                    gatewayConfig.put("userJarPath", systemConfiguration.getSqlSubmitJarPath());
-                    gatewayConfig.put(
-                            "userJarParas",
-                            systemConfiguration.getSqlSubmitJarParas()
-                                    + buildParas(config.getTaskId()));
-                    gatewayConfig.put(
-                            "userJarMainAppClass",
-                            systemConfiguration.getSqlSubmitJarMainAppClass());
+                    Object sqlSubmitJarPath =
+                            Opt.ofBlankAble(gatewayConfig.get("userJarPath"))
+                                    .orElseThrow(
+                                            () ->
+                                                    new DinkyException(
+                                                            "application 模式支持需要在 注册中心->集群管理->集群配置管理 填写jar路径。"));
+                    gatewayConfig.put("userJarParas", buildParas(config.getTaskId()));
+                    gatewayConfig.put("userJarMainAppClass", "org.dinky.app.MainApp");
```

Comment:
the user jar must contain this `MainMapp` class?

Reply:
Is not a user jar, it is dinky-app jar